client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.tsx

```diff
@@ -71,3 +83,54 @@ export function areLaunchpadTasksCompleted(
 
 	return lastTask?.completed;
 }
+
+export function useMaybeSwitchThemeAndUpdateChecklist(
+	checklist: LaunchpadChecklist | null | undefined,
+	siteSlug: string | null
+): LaunchpadChecklist | null | undefined {
+	const site = useSelector( ( state ) => getSite( state, siteSlug ) );
+	const { selectedDesign } = useSelect(
+		( select ) => ( {
+			selectedDesign: ( select( ONBOARD_STORE ) as OnboardSelect ).getSelectedDesign(),
+		} ),
+		[]
+	);
+
+	const reduxDispatch = useReduxDispatch();
+
+	const currentSitePlan = useSelector( ( state ) => {
+		if ( ! site ) {
+			return null;
+		}
+		return getSitePlan( state, site.ID );
+	} );
+
+	const { setDesignOnSite } = useDispatch( SITE_STORE );
+	const { setSelectedDesign } = useDispatch( ONBOARD_STORE );
+
+	const { data: defaultDesign } = useStarterDesignBySlug( 'twentytwentyfour' );
+	const hasPaidDesign =
+		( selectedDesign?.design_tier && selectedDesign?.design_tier !== 'free' ) || false;
+	if ( ! hasPaidDesign || ! currentSitePlan?.is_free ) {
+		return checklist;
+	}
+
+	updateLaunchpadSettings( site?.ID || '', {
+		checklist_statuses: { design_completed: false },
+	} );
+
+	setDesignOnSite( site?.ID, defaultDesign, {
+		styleVariation: defaultDesign?.style_variations?.[ 0 ],
+	} ).then( ( theme: ActiveTheme ) => {
+		return reduxDispatch( setActiveTheme( site?.ID || -1, theme ) );
+	} );
+
+	setSelectedDesign( defaultDesign );
```

Comment:
Are we OK with these running on render? I haven't tested the code but it seems to me that we will dispatch this action and the promise on every render cycle because they're not within an event handler or effect.